RollbarNotifier/Tests/RollbarNotifierTests-ObjC/RollbarAulTests.m

```diff
@@ -0,0 +1,414 @@
+//    NSPredicate *levelPredicate = [NSPredicate predicateWithFormat:@"level == %@", [NSNumber numberWithUnsignedInteger:OS_LOG_TYPE_DEFAULT] ];
+//    //NSPredicate *predicate = [NSCompoundPredicate andPredicateWithSubpredicates:@[levelPredicate, subsystemPredicate, categoryPredicate]];
+//    NSPredicate *predicate = [NSCompoundPredicate andPredicateWithSubpredicates:@[subsystemPredicate, categoryPredicate]];
+//
+//    //[NSThread sleepForTimeInterval:1.0f];
+//
+//    OSLogEnumerator *logEnumerator = [osLogStore entriesEnumeratorWithOptions:0
+//                                                                     position:logPosition
+//                                                                    predicate:predicate
+//                                                                        error:&error];
+//    if (nil != error) {
+//        NSLog(@"AUL ERROR: %@", error);
+//    }
+//
+//    //[NSThread sleepForTimeInterval:1.0f];
+//
+//    //NSUInteger count = logEnumerator.allObjects.count;
+//    int count = 0;
+//    for (OSLogEntryLog *entry in logEnumerator) {
+//
+//        NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+//        [formatter setDateFormat:@"YYYY-MM-dd 'at' HH:mm:ss.SSSSSSXX"];
+//        NSLog(@">>>>> %@: %@", [formatter stringFromDate:entry.date], entry.composedMessage);
+//        count++;
+//    }
+//    NSLog(@"LEGACY Total AUL entries: %d", count);
+    
+// NEW MORE STRUCTURED IMPLEMENTATION:
+    NSError *error = nil;
+    OSLogStore *logStore = [OSLogStore localStoreAndReturnError:&error];
+    if (nil != error) {
+        NSLog(@"AUL ERROR: %@", error);
+    }
+
+    OSLogPosition *logPosition = [logStore positionWithDate:date];
+
+    OSLogEnumerator *logEnumerator = [self buildAulLogEnumeratorWithinLogStore:logStore
+                                                             staringAtPosition:logPosition
+                                                                  forSubsystem:@"com.your_company.your_subsystem"
+                                                                andForCategory:@"your_category_name"];
+    int count = [self processLogEntries:logEnumerator];
+    NSLog(@"Total AUL entries: %d", count);
+    XCTAssertEqual(count, 4);
+}
+
+#pragma mark - RollbarNotifier RollbarAulEntrySnapper
+
+// make it public
+- (void)captureOSLogEntry:(nullable OSLogEntry *)entry
+             intoSnapshot:(nonnull NSMutableDictionary<NSString *, id> *)snapshot {
+    
+    if (nil == entry) {
+        return;
+    }
+    
+    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+    [formatter setDateFormat:@"YYYY-MM-dd 'at' HH:mm:ss.SSSSSSXX"];
+
+    snapshot[@"date"] = [formatter stringFromDate:entry.date];
+    snapshot[@"className"] = entry.className;
+    snapshot[@"composedMessage"] = entry.composedMessage;
+    //snapShot[@""] = entry.storeCategory;
```

Comment:
this can probably be removed

Reply:
That is, for now, to keep as a reminder that there is an extra attribute that we may or may not want to consider capturing. In general, this PR is an experimental playground that will be replaced with unit test implementation that would actually be testing production level AUL integration once it is productized.